processing/src/main/java/org/apache/druid/query/filter/DruidPredicateFactory.java

```diff
@@ -27,6 +27,13 @@
 {
   Predicate<String> makeStringPredicate();
 
+  default Predicate<Object> makeObjectPredicate()
+  {
+    // default to try to use string predicate;
```

Comment:
currently I don't think it should be possible because only expression filter will make a matcher factory with object matching predicate and only for string column types, but i guess cast exceptions if an object selector of a complex column somehow got in here from future code, see other comment about thinking a bit further about this 😅

Reply:
(see next comment https://github.com/apache/druid/pull/10613#discussion_r558127972)